asana-helpers-markdown.js

```diff
@@ -119,14 +135,69 @@
     return null;
   }
 
+  const waitForSuccessNotification = () => (new Promise((resolve, _reject) => {
+    const observer = new MutationObserver((mutations) => {
+      const addedNodes = getAddedNodes(mutations);
+
+      const [notification] = queryAll(addedNodes, '.ToastNotificationContent');
+      if (notification) {
+        observer.disconnect();
+        console.debug('=>> Found notification.', { notification });
+        return resolve(notification);
+      }
+      console.debug('=>> Mutation, but notification?', { mutations });
+    });
+
+    const notificationsDiv = document.querySelector('.ToastManager');
+    console.debug('=>> Observing…', notificationsDiv);
+    observer.observe(notificationsDiv, { subtree: true, childList: true });
+  }));
+
+  const getCurrentTaskUrlFromCopyLinkButton = async () => {
+    const observerWaiting = waitForSuccessNotification();
+    document.querySelector('.TaskPaneToolbar-copyLinkButton,.ConversationToolbar-copyLinkButton').click();
+
+    console.info('=>> Waiting for copied link…');
+    await observerWaiting;
+    console.info('=>> Reading copied link…');
+    const href = await navigator.clipboard.readText();
+    console.debug('=>> Read:', { href }); // TODO: should we verify that the href is actually an URL pointing to app.asana.com?
+    return href;
+  }
+
+  const getCurrentTaskUrlFromPage = async () => {
+    const taskPane = document.querySelector('.TaskPane[data-task-id]');
+    if (taskPane) {
+      const taskId = taskPane.getAttribute('data-task-id');
+      return `https://app.asana.com/0/0/${taskId}/f`;
+    }
+
+    const conversationLink = document.querySelector('.ConversationPane .ConversationTitleAndContext-title a[href]')
+    if (conversationLink) return conversationLink.href;
+
+    // Fallback to Asana's "Copy Link" button.
+    return await getCurrentTaskUrlFromCopyLinkButton();
+  }
+
   // Get a Markdown link for the current task formatted as [Task Title](Link)
-  function getMDLink() {
-    var text = document.getElementsByClassName('TaskPaneToolbarAnimation-title')[0].innerText;
-    var location = window.location;
+  const getMDLink = async (ev) => {
+    console.debug('=>> Click:', ev);
+    const href = await getCurrentTaskUrlFromPage();
 
-    var markdown = `[${text}](${location})`;
+    if (ev.metaKey || ev.ctrlKey || ev.altKey || (ev.button === 1)) { // meta for MacOS, ctrl for Windows, middle click, or alt for good measure
```

Comment:
My source for `button === 1` being the middle button is this: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button